### PR TITLE
Precommit write s3 delete

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [x] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7158 

## What this PR does / why we need it:
1. Add UT 
2. Fixed some bugs.

### CODE REVIEW

File: pkg/vm/engine/tae/catalog/block.go [hunk 1]
```diff
@@ -72,10 +72,11 @@ func NewBlockEntryWithMeta(
 		segment:       segment,
 		state:         state,
 	}
+	e.MetaBaseEntry.CreateWithTxnAndMeta(txn, metaLoc, deltaLoc)
 	if dataFactory != nil {
 		e.blkData = dataFactory(e)
 	}
-	e.MetaBaseEntry.CreateWithTxnAndMeta(txn, metaLoc, deltaLoc)
+	//e.MetaBaseEntry.CreateWithTxnAndMeta(txn, metaLoc, deltaLoc)
```
Comment (fengttt) [inline pkg/vm/engine/tae/catalog/block.go:79]: Is moving this line up intentional?   Add a real comment useful comment and remove this line.

**Review by fengttt — COMMENTED** [review]
What is the s3 delete file format, where is it stored?   I would like to see a doc on this, can you give me a pointer?

File: pkg/vm/engine/tae/catalog/block.go (hunk 1 above)
Comment (triump2020) [reply pkg/vm/engine/tae/catalog/block.go:79]: Yes , it's intentional.

**Comment by JackTan25** [comment]
> delete will write only one col, it's the rowId, the rowId is generated by CN. And this will be sent to DN

**Review by nnsgmsone — APPROVED** [review]

**Review by jiangxinmeng1 — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Review by sukki37 — APPROVED** [review]

**Pull request merged**